New chronosuit teleport mechanics

JJRcop opened the pull request:
### Revised edition of chronosuit teleport mechanics:
- You no longer continuously teleport to the camera while moving, you must stop moving for a few moments or press "teleport now" to teleport
- Your teleport is a static amount of time when phasing out, but it takes an amount of time to phase in depending on how far you travel, which is approximately 1 second per 5 tiles
- Your suit still deactivates when strong EMPed like normal, but if you are EMPed while you are in the middle of teleporting, the suit deactivates, but you are also electrocuted for 35 damage
- While using the camera to maneuver to the target location, a ghost icon of you sticks to the center of the screen to help show you which tile you are teleporting to
- New phasing animation

---
#### Demo video: https://youtu.be/VZxjzciobLA

Commit:
New chronosuit teleport mechanics

PKPenguin321 commented:
i love the 10,000 emps in that demo video
also, admins should use these more

JJRcop commented on code/modules/clothing/spacesuits/chronosuit.dm:
```diff
@@ -82,52 +109,60 @@
+		phase_underlay.pixel_y = user.pixel_y
+
+		user.animate_movement = NO_STEPS
+		user.next_click = world.time + (8 + phase_in_ds)
```
i realize this is bad i'm working on it

Bawhoppen commented:
bigtime nerf to chrono-legionairres, maybe they will see some use now

RemieRichards commented on code/modules/clothing/spacesuits/chronosuit.dm:
```diff
@@ -82,52 +109,60 @@
+		phase_underlay.pixel_x = user.pixel_x
+		phase_underlay.pixel_y = user.pixel_y
+
+		user.animate_movement = NO_STEPS
```
Wow hey wtf are you doing here.
P.sure this doesn't do what you think it does.

JJRcop replied:
making it so you don't slide if you teleport to an adjacent tile, it does do exactly what I think it does

JJRcop commented:
@Bawhoppen Yep, that's the plan!

Commit:
Chronosuit uses next_move instead of next_click
face_atom doesn't do anything if notransform is nonzero
Added Timeline Eradication Agent outfit

Commit:
Merge remote-tracking branch 'upstream/master' into chronosuit_phase_better

JJRcop commented:
I fixed the next_click stuff and merge conflicts

JJRcop commented:
Broken atm, "teleport now" button needs to be modified

Commit:
Merge remote-tracking branch 'upstream/master' into chronosuit_phase_better

JJRcop commented:
Fixed it

Commit:
Added changelog

JJRcop commented:
Aran brought up some points that I should fix, don't merge this yet

JJRcop commented:
Had some downtime, will work on tomorrow or weekend.

Commit:
Chronosuit performance/saneness adjustments

Commit:
Merge remote-tracking branch 'upstream/master' into chronosuit_phase_better

JJRcop commented:
Mokay I think that's it.

RemieRichards commented on code/modules/clothing/spacesuits/chronosuit.dm:
```diff
@@ -25,19 +25,27 @@
 	action_button_name = "Toggle Chronosuit"
 	armor = list(melee = 60, bullet = 60, laser = 60, energy = 60, bomb = 30, bio = 90, rad = 90)
 	var/list/chronosafe_items = list(/obj/item/weapon/chrono_eraser, /obj/item/weapon/gun/energy/chrono_gun)
+	var/list/hands_nodrop_states
```
a list for 2 values is silly.
use a bitfield.

JJRcop replied:
ah good idea

Commit:
Changes chronosuit hands_nodrop_states to a bitfield from a list

JJRcop commented:
Dee dum, oh, fixed!
